src/app/components/activity/feed/post/AppActivityFeedPostHeader.vue

```diff
@@ -0,0 +1,135 @@
+<script lang="ts" setup>
+import { computed, PropType } from 'vue';
+import { FiresidePost } from '../../../../../_common/fireside/post/post-model';
+import { Screen } from '../../../../../_common/screen/screen-service';
+import { AppTimeAgo } from '../../../../../_common/time/ago/ago';
+import AppUserCardHover from '../../../../../_common/user/card/hover/hover.vue';
+import AppUserFollowWidget from '../../../../../_common/user/follow/widget.vue';
+import AppUserAvatar from '../../../../../_common/user/user-avatar/user-avatar.vue';
+import AppUserVerifiedTick from '../../../../../_common/user/verified-tick/verified-tick.vue';
+import { ActivityFeedView } from '../view';
+import AppActivityFeedPostTime from './time/time.vue';
+
+const props = defineProps({
+	post: {
+		required: true,
+		type: Object as PropType<FiresidePost>,
+	},
+	feed: {
+		default: null,
+		type: (null || Object) as PropType<ActivityFeedView | null>,
+	},
+	showPinned: {
+		default: false,
+		type: Boolean,
+	},
+	showDate: {
+		default: false,
+		type: Boolean,
+	},
+	dateLink: {
+		default: null,
+		type: String as PropType<string | null>,
+	},
+});
+
+const user = computed(() => props.post.displayUser);
+const overlay = computed(() => !!props.post.background);
+const feed = computed(() => props.feed);
+
+const game = computed(() => props.post.game);
+const gameUrl = computed(() => game.value?.getUrl());
+
+const shouldShowFollow = computed(() => {
+	// Don't show follow for game posts. Only for user posts.
+	if ((feed.value && !feed.value.shouldShowFollow) || props.post.game) {
+		return false;
+	}
+
+	if (props.post.user.blocked_you) {
+		return false;
+	}
+
+	// Don't show follow if already following.
+	if (!user.value || user.value.is_following) {
+		return false;
+	}
+
+	return true;
+});
+</script>
+
+<template>
+	<div v-if="user" class="-container-theme -header">
+		<div class="-header-content">
+			<AppUserCardHover :user="user" :disabled="feed && feed.shouldShowUserCards">
+				<div class="-header-avatar">
+					<div class="-header-avatar-inner">
+						<AppUserAvatar :user="user" />
+					</div>
+				</div>
+			</AppUserCardHover>
+
+			<div class="-header-byline">
+				<div class="-header-byline-name" :class="{ '-overlay-text': overlay }">
+					<strong>
+						<router-link
```

Comment:
Convert all router-link and router-view stuff into RouterLink and RouterView and import them. It gives proper type checking for props then